test/e2e/framework/metrics/metrics_proxy.go

```diff
@@ -68,8 +68,8 @@ func SetupMetricsProxy(c clientset.Interface) error {
 				})
 			}
 		}
-		if len(foundComponents) != 2 {
-			klog.Infof("Only %d components found. Will retry.", len(foundComponents))
+		if len(foundComponents) < 2 {
```

Comment:
In a 3 node control plane, this value is 6, so we poll for 5 minutes unnecessarily.